### PR TITLE
Latest Instagram Posts: Support for Multiple User Accounts

### LINKED ISSUE #15848: Latest Instagram Posts Block: Allow multiple tokens per user
Currently, the Latest Instagram Posts block only allows 1 Instagram token per user, shared across all WPCOM sites.

Let's see if it's possible to allow them to use multiple tokens instead.

- [ ] On disconnection, don't delete the token. Instead, just remove it from the block's attributes.
Users will still be able to permanently delete the token on Instagram or in the Marketing/Connections page in Calypso.

- [ ] Let users connect different Instagram accounts to their WPCOM users. This might need some Keyring investigation to see if it's possible (it should), and in the block's endpoints to see if there are any changes needed.

### PR DESCRIPTION
**Pull request by Copons**
Fixes #15848

#### Changes proposed in this Pull Request:

* Add support for multiple user accounts in the Latest Instagram Posts block.
  * Users with active Instagram tokens, can reuse them straight away.
  * Users without, will need to connect to Instagram through the usual popup.
* Disconnecting the block won't delete the token anymore, but simply remove it from the block attributes.
  * To permanently delete the token, users will need to head to Marketing/Connections in Calypso, or do it directly from Instagram.
* Remove 2 unnecessary endpoints (`GET /access-token` and `DELETE /delete-access-token`), and corresponding helper code.

#### Screenshots

cc @davemart-in and @olesyabrk for suggestions about the placeholder copy, especially the account selector, and the disconnection button in the block sidebar.

User without connected Instagram accounts:
Screenshot 2020-05-19 at 19 28 26

User with connected Instagram accounts:
Screenshot 2020-05-20 at 11 48 30

Disconnection button:
Screenshot 2020-05-20 at 11 48 23

#### Testing instructions:

WPCOM prerequisites: apply D43668-code and D43669-code (hopefully it patches cleanly).

- Smoke test the Latest Instagram Posts block, and make sure the UX feels good.
- Try connecting more than one Instagram account.
- Make sure connecting new accounts is reflected in the account selector.
- Make sure that after disconnecting a block, the previous account is automatically selected in the account selector.
- Make sure blocks connected to different accounts are visible on the front end to visitors.

#### Proposed changelog entry for your changes:

* N/A

### CODE REVIEW

**Comment by olesyabrk** [comment]
@Copons here are some suggestions on copy. 

image

- Remove the widget description, it's self-explanatory and is duplicated in the sidebar. 
- Let's align the content to the header, not to the icon 

image

**Comment by Copons** [comment]
Thanks @olesyabrk for the suggestions!
I've updated some things, but I've decided to leave others as they are for consistency.

---

Placeholder when user already has some connections:
- Removed unnecessary instructions.
- Set the font weight of the radios to normal.

Screenshot 2020-05-20 at 11 48 30

Although, when the user doesn't have any connections, I've chosen to leave the "connect to start sharing" instructions, in order to avoid having a "stray" button.
Also, I think we should leave the placeholder content aligned to the left, as it's what Gutenberg Core does (try with an Image block for example).

---

Sidebar account settings:
- Aligned the account name to the left.
- Updated the disconnection copy.

Screenshot 2020-05-20 at 11 48 23

What I've not touched:
- The account name is an external link (opening in a new tab, outside the current site), so it needs the external icon.
- The disconnection doesn't navigate away, but simply "resets" the block to the placeholder. The arrow icon seems a bit out of place to me. 🤔

File: extensions/blocks/instagram-gallery/edit.js [hunk 1]
```diff
@@ -183,32 +239,17 @@ const InstagramGalleryEdit = props => {
 			{ showSidebar && (
 				<InspectorControls>
 					<PanelBody title={ __( 'Account Settings', 'jetpack' ) }>
-						<PanelRow>
-							<span>{ __( 'Account', 'jetpack' ) }</span>
+						<div>
+							{ __( 'Account:', 'jetpack' ) }{ ' ' }
 							<ExternalLink href={ `https://www.instagram.com/${ instagramUser }/` }>
 								@{ instagramUser }
 							</ExternalLink>
```
Comment (jeherve) [inline extensions/blocks/instagram-gallery/edit.js:246]: I am not sure this is going to work on RTL languages, maybe this should be one translatable string with a placeholder.

Comment (Copons) [reply extensions/blocks/instagram-gallery/edit.js:246]: Ooh, makes sense, thanks!

Comment (Copons) [reply extensions/blocks/instagram-gallery/edit.js:246]: No, wait, now I remembered why I did it in two columns with `PanelRow` before: Guten `i18n` functions don't support React components.

I'll revert this change back to the two columns version. 🤦

**Comment by scruffian** [comment]
Screenshot 2020-05-21 at 12 01 53

I wonder if we could change the button text here to explain what is happening more clearly... Maybe "Loading your connections…"?

**Comment by scruffian** [comment]
The highlighted state of the radio button looks weird:
Screenshot 2020-05-21 at 12 06 33

Also, when I click "Add a new account" it connects to the account I am currently logged in with, which is the one that's already connected. I know this is a limitation we have but I worry that a lot of users will struggle to understand this. Not sure how we can get around it without adding some messaging though....

**Comment by scruffian** [comment]
After I disconnect the block, the connection button is disabled for a moment. I'm not sure why this is necessary:
May-21-2020 12-17-42

**Comment by Copons** [comment]
> The highlighted state of the radio button looks weird:
> Screenshot 2020-05-21 at 12 06 33

🤔 Weird, I haven't touched its style at all (just removed the bold, but I don't think it's related).

> 
> Also, when I click "Add a new account" it connects to the account I am currently logged in with, which is the one that's already connected. I know this is a limitation we have but I worry that a lot of users will struggle to understand this. Not sure how we can get around it without adding some messaging though....

Yeah, totally agree with this.
Considering there's no way to change IG account from the IG oAuth screen, we are pretty much forced to add a message about it.
Worst thing is, that it's sort of a complicated one to convey in simple words.
What about showing this line under the "add account" option, only when it's selected?

"If you're currently logged in to Instagram on this device, you might need to log out of it first."

Maybe we could also add a "Learn more" link pointing to the block's support page.

> After I disconnect the block, the connection button is disabled for a moment. I'm not sure why this is necessary:
> May-21-2020 12-17-42

It's loading/refreshing the connections, in the off chance they've changed meanwhile.
It'll become more clear when I'll update the button with the "loading" label you suggested earlier.

File: extensions/blocks/instagram-gallery/editor.scss [hunk 2]
```diff
@@ -1,5 +1,18 @@
 @import './view.scss';
 
+.wp-block-jetpack-instagram-gallery .components-placeholder {
+	.components-radio-control {
+		margin-bottom: $default-block-margin;
+		label {
+			font-weight: normal;
```
Comment (scruffian) [inline extensions/blocks/instagram-gallery/editor.scss:7]: I tried removing this and it looks the same. Is it for a different version of Gutenberg?

Comment (Copons) [reply extensions/blocks/instagram-gallery/editor.scss:7]: Interesting!
If I remove that line, it looks like this to me:

Screenshot 2020-05-21 at 17 49 03

I've tested on Guten 8.1 and 8.0, and with Guten deactivated, and in all cases I see this style in the dev tools:
```css
.editor-styles-wrapper label {
  font-weight: 600;
}
```

(Actually, I'm not entirely sure I tested on 8.0. I just updated Guten this morning after a while, and it's 8.1 now. Maybe it was a 8.1 beta before.)

Comment (scruffian) [reply extensions/blocks/instagram-gallery/editor.scss:7]: Maybe it's a theme thing? Doesn't hurt to leave it in :)

**Comment by Copons** [comment]
@jeherve apologies for the push, but I discovered that some themes (2020 in this case) set the `p` font family with more specificity than the placeholder fieldset.
I addressed this in 6a91a1f.

| Before | After |
| - | - |
| Screenshot 2020-05-22 at 12 24 03 | Screenshot 2020-05-22 at 12 24 41 |

Since this dismissed the reviews, I'll also take the chance of merging #15887 first and rebase this.

**Comment by Copons** [comment]
r207896-wpcom

**Pull request merged**